staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -247,15 +245,13 @@ type mutateObjectUpdateFunc func(obj, old runtime.Object) error
 // moved into this type.
 type patcher struct {
 	// Pieces of RequestScope
-	namer           ScopeNamer
-	creater         runtime.ObjectCreater
-	defaulter       runtime.ObjectDefaulter
-	typer           runtime.ObjectTyper
-	unsafeConvertor runtime.ObjectConvertor
-	resource        schema.GroupVersionResource
-	kind            schema.GroupVersionKind
-	subresource     string
-	dryRun          bool
+	namer       ScopeNamer
+	resource    schema.GroupVersionResource
+	kind        schema.GroupVersionKind
+	subresource string
+	dryRun      bool
+
+	objectInterfaces admission.ObjectInterfaces
```

Comment:
the set of things this patcher needs happens to intersect with the admission interface, but I'd probably leave this as alone, rather than require the patcher by given the object interface instance. I could imagine future things added to the admission interface that the patcher didn't care about

Reply:
I reverted changes to this patcher (except passing over objectInterfaces to admissions). I also removed "GetObjectSafeConvertor" that was only used in this patcher.